Feature mobile markup

AdreyO opened the pull request:
Я уже совместил файлы через гит и проверил, сайт работает корректно
НО... при создании комита этой ветки в мейн - пишет, что автоматическое слияние невозможно, как решить?

Mardoyan-Gayane committed:
Add files via upload

AdreyO committed:
feat-welcome #2

исправил теги li

Mardoyan-Gayane committed:
mobile

Mardoyan-Gayane committed:
Merge branch 'feature-mobile-markup' of https://github.com/AdreyO/Frontend-app_2 into feat-sightseens

Mardoyan-Gayane committed:
xren

AdreyO committed:
itog section "welcome"

ПК + мобильная версия
- анимация бургера
- анимация стрелки пролистывания
- анимация шапки
- текст обрезается

AdreyO committed:
Merge branch 'feature-mobile-markup' into feat-sightseens

AdreyO committed:
Update styles.css

AdreyO committed:
Merge pull request #4 from AdreyO/feat-sightseens

Feat sightseens and feat-welcome

AdreyO committed:
Update README.md

AdreyO committed:
Merge branch 'main' into feature-mobile-markup

AdreyO commented:
> Я уже совместил файлы через гит и проверил, сайт работает корректно НО... при создании комита этой ветки в мейн - пишет, что автоматическое слияние невозможно, как решить?

Я РАЗОБРАЛСЯ, ТЕПЕРЬ ВСЕ РАБОТАЕТ

AdreyO committed:
itog

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,224 @@
+    </section>
+
+    <section class="tours">
+        <h1 class="tours__title">
```
h1 только один может быть на странице

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,224 @@
+        <div class="tours__list">
+            <div class="tour">
+                <div class="">
+                    <h2 class="tour__title">Название</h2>
```
Лучше h3

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,224 @@
+            </div> 
+
+            <div class="tour">
+                <div class="">
```
Зачем div с пустым классом?

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,224 @@
+    </section>
+ 
+    <section class="contacts">
+        <h1 class="contacts__title">
```
h1 только 1

MikeBazhenov commented on js/app.js:
```diff
@@ -0,0 +1,30 @@
+$(document).ready(function () {
```
Без jquery надо

Mardoyan-Gayane committed:
Update index.html

AdreyO committed:
itog v2

- из js убран jQuery
-исправлен index.html

AdreyO committed:
Merge branch 'feature-mobile-markup' of https://github.com/AdreyO/Frontend-app_2 into feature-mobile-markup

AdreyO commented:
@MikeBazhenov 
Мы исправили вышеуказанные замечания, проверьте пожалуйста

MikeBazhenov commented on js/app.js:
```diff
@@ -0,0 +1,33 @@
+document.addEventListener('DOMContentLoaded', function(){
+
+    let burger = document.querySelector('.header__burger');
```
Нужно использовать const, переменная не перезаписывается

MikeBazhenov commented on js/app.js:
```diff
@@ -0,0 +1,33 @@
+document.addEventListener('DOMContentLoaded', function(){
+
+    let burger = document.querySelector('.header__burger');
+    let menu = document.querySelector('.header__menu');
```
const

MikeBazhenov commented on js/app.js:
```diff
@@ -0,0 +1,33 @@
+
+    let burger = document.querySelector('.header__burger');
+    let menu = document.querySelector('.header__menu');
+    let body = document.querySelector('body');
```
const

MikeBazhenov commented on js/app.js:
```diff
@@ -0,0 +1,33 @@
+    let burger = document.querySelector('.header__burger');
+    let menu = document.querySelector('.header__menu');
+    let body = document.querySelector('body');
+    let header = document.querySelector('.header');
```
const

MikeBazhenov commented on js/app.js:
```diff
@@ -0,0 +1,33 @@
+        burger.classList.toggle('active')
+        menu.classList.toggle('active')
+        body.classList.toggle('lock')
+        if(header.classList.contains('header_scroll')){
```
header.classList.toggle('header_scroll')

Зачем if если до этого пользуешься toggle?

AdreyO replied:
toggle для header - это просто недоглядел, изначально было так
а if необходим, чтоб при открытии header burger  убирался цвет header__scroll, иначе будет наложение темной полоски в меню
ибо в верхнем положении header прозрачный, а при скролле он темный

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+    </section>
+
+    <section id="tours" class="tours">
+        <h2 class="tours__title">
```
h3

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+    </header>
+
+    <section id="welcome" class="welcome">
+        <div class="welcome-container">
```
```suggestion
        <header class="welcome-container">
```

AdreyO replied:
Если это предложение заменить мой код, то я не понимаю зачем и почему так?

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+        <div class="welcome-container">
+            <h1 class="welcome-title">Хакасия</h1>
+            <p class="welcome-subtitle">Туристическое Агенство Республики Хакасия</p>
+        </div>
```
```suggestion
        </header>
```

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+            <div class="header__body">
+                <a class="header__logo"></a>
+                <div class="header__burger">
+                    <span>  </span>
```
```suggestion
                    <span> </span>
```

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+    <header id="header" class="header">
+        <div class="container">
+            <div class="header__body">
+                <a class="header__logo"></a>
```
```suggestion
                <a class="header__logo" href="/"></a>
```

Mardoyan-Gayane committed:
Update index.html

AdreyO committed:
itog v3

-правки js

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+                    <p class="block__text-item">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur pretium orci at rutrum pellentesque. Vivamus id nunc at ante vulputate egestas. Sed varius elit ornare condimentum fermentum. Integer mattis pretium tortor dictum tristique. Donec in congue mauris. Quisque vitae nisi ultricies, efficitur odio vel, porta nulla. Donec porta, mauris nec ...
+                </div>
+            </div>
+            <div class="block__item" id="down">
```
.block-item:nth-child(even) . block__boximg {
}

.block-item:nth-child(even) . block__boxtext {
}

AdreyO replied:
добавлено

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+                    <img class="block__img" src="img/sightseens.png">
+                </div>
+                <div class="block__boxtext block__item-order0">
+                    <p  class="block__name-item">Приисковый</p>
```
h4

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+                    <img class="block__img" src="img/sightseens.png">
+                </div>
+                <div class="block__boxtext">
+                    <p class="block__name-item">Сундуки</p>
```
h4

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+        <div class="tours__list">
+            <div class="tour">
+                <div class="container">
+                    <h3 class="tour__title">Название</h3>
```
h4

Mardoyan-Gayane replied:
Исправлено

MikeBazhenov commented on index.html:
```diff
@@ -0,0 +1,208 @@
+                </div>
+            </div>
+        </div>
+    </section>
```
Нужно добавить кнопку

Mardoyan-Gayane replied:
Добавлено

AdreyO committed:
itog v4

+ анимация крестика (уплывает вниз)
+ реализован nth-child(even)
+ js изменен на "как надо"
+ функция для выделения пункта меню, в зависимости от положения стр.

AdreyO committed:
itog v4

Mardoyan-Gayane committed:
Update index.html

Mardoyan-Gayane committed:
Update app.js

Mardoyan-Gayane committed:
Update styles.css

Mardoyan-Gayane committed:
Update index.html

Mardoyan-Gayane committed:
Update styles.css

Mardoyan-Gayane committed:
Update styles.css

Mardoyan-Gayane committed:
Update index.html

Mardoyan-Gayane committed:
Update styles.css

Mardoyan-Gayane committed:
Update styles.css

Mardoyan-Gayane committed:
Update styles.css

Mardoyan-Gayane committed:
Update index.html

Mardoyan-Gayane committed:
Update app.js

MikeBazhenov commented on js/app.js:
```diff
@@ -0,0 +1,88 @@
+    var windowY = window.pageYOffset+500
+    console.log(windowY,welcome.bottom,attractions.top);
+    if (windowY >= welcome.top-body.top && windowY < attractions.top-body.top){
+        link[1].style.cssText='text-decoration:underline;';
```
Плохо менять стили через JS можно было обойтись обычным добавлением класса. Не критично, на будущее.

MikeBazhenov commented:
Принято
